### DISCUSSION TITLE
How to get the coordinates of the image after rotation

### DISCUSSION THREAD

**Discussion by mor-anhln** [chartjs/chartjs-plugin-annotation#935] [Q&A] [answered]
I have a rectangle, after rotating it 90 degrees, the coordinates returned are still the same as the original, how can I get the coordinates after rotation?
![image](https://github.com/user-attachments/assets/c5420120-2623-4ad1-b3a1-de3a1cd7bc65)

**Comment by stockiNail** [comment, marked as answer]
@anhln1998 the plugin  is not exporting the helpers therefore, even if there is the function, you cannot invoked.

Nevertheless, is not an issue for you, you can take it and copy in your project:

https://github.com/chartjs/chartjs-plugin-annotation/blob/e94475cc299fd77dd6c008b7f4bfd4a985df3a6d/src/helpers/helpers.geometric.js#L5-L22

**Reply by mor-anhln** [reply]
@stockiNail 
Thanks for your feedback, after applying the rotated function there has been some progress, but it seems the coordinates are still not very accurate.
![image](https://github.com/user-attachments/assets/ca82a4f8-f14b-4e88-972e-c8b53c7db795)

**Reply by stockiNail** [reply]
can you share the point, center and angle values?

**Reply by mor-anhln** [reply]
@stockiNail 
yes here are my points
[
    {
        "label": "x1",
        "x": 17194.194042259387,
        "y": 16159.178351604034
    },
    {
        "label": "x2",
        "x": 14914.033383944323,
        "y": 6626.810839287125
    },
    {
        "label": "y1",
        "x": 12805.805957740615,
        "y": 8840.821648395962
    },
    {
        "label": "y2",
        "x": 15085.966616055677,
        "y": 18373.18916071287
    }
]

centerX: 332.6225446428572
centerY: 195.98571428571427
angle: 90

**Reply by stockiNail** [reply]
@anhln1998 as report in the function doc, the angle must be in RADIANS and not DEGREE.

You can use Chart.js helper, function `toRadians`.

```
export function toRadians(degrees: number) {
  return degrees * (PI / 180);
}
```

**Reply by mor-anhln** [reply]
i did it thank you very much @stockiNail
